pkg/handler/v2/alertmanager/alertmanager_test.go

```diff
@@ -0,0 +1,410 @@
+		ExistingConfigSecret      *corev1.Secret
+		ExistingAPIUser           *apiv1.User
+		ExpectedResponse          apiv2.Alertmanager
+		ExpectedHTTPStatus        int
+	}{
+		{
+			Name:      "scenario 1: create alertmanager",
+			ProjectID: test.GenDefaultProject().Name,
+			ClusterID: test.GenDefaultCluster().Name,
+			ExistingKubermaticObjects: test.GenDefaultKubermaticObjects(
+				test.GenTestSeed(),
+				test.GenDefaultCluster(),
+			),
+			Body: generateRequestBody([]byte(testAlertmanagerConfig)),
+			ExistingAlertmanager: test.GenAlertmanager(test.GenDefaultCluster().Status.NamespaceName,
+				testAlertmanagerConfigSecretName),
+			ExistingConfigSecret: test.GenAlertmanagerConfigSecret(testAlertmanagerConfigSecretName,
+				test.GenDefaultCluster().Status.NamespaceName,
+				[]byte("test")),
+			ExistingAPIUser:    test.GenDefaultAPIUser(),
+			ExpectedHTTPStatus: http.StatusCreated,
+			ExpectedResponse: apiv2.Alertmanager{
+				Spec: apiv2.AlertmanagerSpec{
+					Config: []byte(testAlertmanagerConfig),
+				},
+			},
+		},
+		{
+			Name:      "scenario 2: create alertmanager with invalid request",
+			ProjectID: test.GenDefaultProject().Name,
+			ClusterID: test.GenDefaultCluster().Name,
+			ExistingKubermaticObjects: test.GenDefaultKubermaticObjects(
+				test.GenTestSeed(),
+				test.GenDefaultCluster(),
+			),
+			Body: generateRequestBody([]byte("bad-request")),
+			ExistingAlertmanager: test.GenAlertmanager(test.GenDefaultCluster().Status.NamespaceName,
+				testAlertmanagerConfigSecretName),
+			ExistingConfigSecret: test.GenAlertmanagerConfigSecret(testAlertmanagerConfigSecretName,
+				test.GenDefaultCluster().Status.NamespaceName,
+				[]byte("test")),
+			ExistingAPIUser:    test.GenDefaultAPIUser(),
+			ExpectedHTTPStatus: http.StatusBadRequest,
+		},
+		{
+			Name:      "scenario 3: alertmanager is not found",
+			ProjectID: test.GenDefaultProject().Name,
+			ClusterID: test.GenDefaultCluster().Name,
+			ExistingKubermaticObjects: test.GenDefaultKubermaticObjects(
+				test.GenTestSeed(),
+				test.GenDefaultCluster(),
+			),
+			Body:               generateRequestBody([]byte(testAlertmanagerConfig)),
+			ExistingAPIUser:    test.GenDefaultAPIUser(),
+			ExpectedHTTPStatus: http.StatusInternalServerError,
+		},
+		{
+			Name:      "scenario 4: alertmanager config secret is not found",
+			ProjectID: test.GenDefaultProject().Name,
+			ClusterID: test.GenDefaultCluster().Name,
+			ExistingKubermaticObjects: test.GenDefaultKubermaticObjects(
+				test.GenTestSeed(),
+				test.GenDefaultCluster(),
+			),
+			ExistingAlertmanager: test.GenAlertmanager(test.GenDefaultCluster().Status.NamespaceName,
+				testAlertmanagerConfigSecretName),
+			Body:               generateRequestBody([]byte(testAlertmanagerConfig)),
+			ExistingAPIUser:    test.GenDefaultAPIUser(),
+			ExpectedHTTPStatus: http.StatusInternalServerError,
+		},
+		{
+			Name:      "scenario 5: user john can not get alertmanager that belongs to bob's cluster",
```

Comment:
NIT: `...can not get...` -> `...can not create...`

Reply:
Fixed, thanks!